File: Frontend/src/types/spaceship.ts
export interface SpaceshipContainer {
    spaceship: Spaceship;
    hasSpaceship: boolean;
}

export interface Spaceship {
    ownerId: string
    galaxyVersion: number,
    locatedRadius: number,
    locatedAngleMilliradians: number
}

export interface NamedSpaceship {
    id: string;
    name: string;
    battlePower: number;
}

File: Argonauts.Core/Entity/Player/Spaceship.cs
namespace Argonauts.Core.Entity.Player;

/// <summary>
/// Represents a spaceship entity with owner and location information.
/// </summary>
public class Spaceship
{
    /// <summary>
    /// Unique identifier of the spaceship owner.
    /// </summary>
    public Guid OwnerId { get; init; }

    /// <summary>
    /// Galaxy version when the spaceship was created.
    /// </summary>
    public int GalaxyVersion { get; init; }

    /// <summary>
    /// The current radius coordinate where the spaceship is located.
    /// </summary>
    public int LocatedRadius { get; set; }

    /// <summary>
    /// The current angle coordinate in milliradians.
    /// </summary>
    public int LocatedAngleMilliradians { get; set; }
}